src/utils.ts

```diff
@@ -167,23 +173,57 @@ export async function openDDLEditor(vdbName: string) {
 	}
 }
 
-export function createOrUpdateLocalReferenceFile(refName: string, variableName: string, variableValue: string, refType: string) {
+export async function createOrUpdateLocalReferenceFile(refName: string, variableName: string, variableValue: string, refType: string) {
 	// TODO: create or update the entry reference in a local yaml file for configMap OR secret format
 	// TODO: implement me!
 }
 
-export function doesLocalReferenceFileExist(vdbFile: string, refName: string, refType: string): boolean {
-	// TODO: check if a file exists for the given type and name in the same folder than the vdbFile
-	// TODO: implement me!
-	return false;
+export function getFullReferenceFilePath(vdbFile: string, refName: string): string {
+	return path.join(path.dirname(vdbFile), `${refName}.yaml`);
 }
 
-export function loadPredefinedVariables(vdbFile: string, refName: string, refType: string): Array<Property> {
-	// TODO: load all defined key value pairs from the reference file into an array of property objects -> only name and value will be used
-	// TODO: implement me!
+export async function doesLocalReferenceFileExist(vdbFile: string, refName: string): Promise<boolean> {
+	const refFile: string = getFullReferenceFilePath(vdbFile, refName);
+	return await doesFileExist(refFile);
+}
+
+export async function doesFileExist(file: string): Promise<boolean> {
+	try {
+		await vscode.workspace.fs.stat(vscode.Uri.file(file));
+		return true;
+	} catch (error) {
+		return false;
+	}
+}
```

Comment:
this method is checking if the file if the stat of the file is accessible. it will return false also in case of permission error or if the file is a directory.
Maybe worth modifying the name of it to be more accurate.

Reply:
mentioned on the other PR https://github.com/jboss-fuse/vscode-datavirt/pull/167#discussion_r425910336

Reply:
the mentioned error outputs have been removed

Reply:
I think the naming of the function is fine. One way or the other the user doesn't have access to the file, so it doesn't exist in both cases.